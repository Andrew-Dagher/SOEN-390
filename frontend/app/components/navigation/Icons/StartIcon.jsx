/**
 * @file StartIcon.jsx
 * @description A React Native SVG component for a start icon.
 */

import * as React from "react";
import Svg, { Path } from "react-native-svg";

/**
 * StartIcon component renders a start icon.
 *
 * @component
 * @param {Object} props - Component properties.
 * @returns {JSX.Element} The rendered SVG start icon.
 */
const StartIcon = (props) => (
  <Svg
    xmlns="http://www.w3.org/2000/svg"
    width={14}
    height={18}
    fill="none"
    {...props}
  >
    {/* Start marker path */}
    <Path
      fill="#FFD4D4"
      d="M1.283 17.705c.419 0 .686-.15 1.097-.552l4.503-4.462c.042-.042.075-.075.117-.075.042 0 .075.034.117.075l4.504 4.462c.41.402.678.552 1.096.552.561 0 .912-.435.912-1.004 0-.318-.133-.695-.259-1.038L8.138 1.366C7.871.629 7.478.295 7 .295c-.477 0-.87.334-1.138 1.071L.63 15.663c-.125.343-.26.72-.26 1.038 0 .569.352 1.004.913 1.004Z"
    />
  </Svg>
);

export default StartIcon;
