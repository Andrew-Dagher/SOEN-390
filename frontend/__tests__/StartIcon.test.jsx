/**
 * @file StartIcon.test.jsx
 */

import React from "react";
import { render } from "@testing-library/react-native";
import StartIcon from "../app/components/navigation/Icons/StartIcon";

describe("StartIcon component", () => {
  it("renders without crashing", () => {
    // Render the component
    const { getByTestId } = render(<StartIcon testID="start-icon-svg" />);
    // Confirm the SVG is in the tree with the expected testID
    const svgElement = getByTestId("start-icon-svg");

    // Validate key props
    expect(svgElement.props.width).toBe(14);
    expect(svgElement.props.height).toBe(18);
    expect(svgElement.props.fill).toBe("none");
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(<StartIcon />);
    expect(toJSON()).toMatchSnapshot();
  });
});
