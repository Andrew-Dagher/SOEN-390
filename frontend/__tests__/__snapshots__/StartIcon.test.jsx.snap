// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartIcon component matches the snapshot 1`] = `
<RNSVGSvgView
  bbHeight={18}
  bbWidth={14}
  fill="none"
  focusable={false}
  height={18}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 18,
        "width": 14,
      },
    ]
  }
  width={14}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M1.283 17.705c.419 0 .686-.15 1.097-.552l4.503-4.462c.042-.042.075-.075.117-.075.042 0 .075.034.117.075l4.504 4.462c.41.402.678.552 1.096.552.561 0 .912-.435.912-1.004 0-.318-.133-.695-.259-1.038L8.138 1.366C7.871.629 7.478.295 7 .295c-.477 0-.87.334-1.138 1.071L.63 15.663c-.125.343-.26.72-.26 1.038 0 .569.352 1.004.913 1.004Z"
      fill={
        {
          "payload": 4294956244,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
